python/ray/_common/test_utils.py

```diff
@@ -1,39 +1,436 @@
+    ), "Should load TemporaryDirectory class"
+    print("✓ Standard library class loading test passed")
+
+    # Test 4: Test error handling
+    try:
+        ray_load_class("invalid")
+        assert False, "Should raise ValueError"
+    except ValueError:
+        print("✓ Invalid path error handling test passed")
+
+    try:
+        ray_load_class("nonexistent_module.SomeClass")
+        assert False, "Should raise ImportError"
+    except (ImportError, ModuleNotFoundError):
+        print("✓ Nonexistent module error handling test passed")
+
+    try:
+        ray_load_class("os.NonexistentClass")
+        assert False, "Should raise AttributeError"
+    except AttributeError:
+        print("✓ Nonexistent attribute error handling test passed")
+
+    print("All load_class tests passed!")
+
+
+def _test_with_function(func, name):
+    """Helper function to test directory creation with a given function."""
+    # Create a temporary directory for testing
+    with tempfile.TemporaryDirectory() as temp_dir:
+        # Test 1: Create a new directory
+        test_dir = os.path.join(temp_dir, "test_dir")
+        func(test_dir)
+        assert os.path.exists(test_dir), "Directory should be created"
+        assert os.path.isdir(test_dir), "Path should be a directory"
+        print(f"✓ {name} Test 1 passed: Directory creation")
+
+        # Test 2: Create directory that already exists (should not raise error)
+        func(test_dir)  # Should work without error
+        assert os.path.exists(test_dir), "Directory should still exist"
+        print(f"✓ {name} Test 2 passed: Existing directory handling")
+
+        # Test 3: Create nested directory structure
+        nested_dir = os.path.join(temp_dir, "nested", "deep", "structure")
+        func(nested_dir)
+        assert os.path.exists(nested_dir), "Nested directory should be created"
+        print(f"✓ {name} Test 3 passed: Nested directory creation")
+
+        # Test 4: Test with ~ expansion (home directory)
+        fake_home = os.path.join(temp_dir, "fake_home")
+        os.makedirs(fake_home, exist_ok=True)
+
+        # Mock the expanduser for this test
+        original_expanduser = os.path.expanduser
+        os.path.expanduser = (
+            lambda path: path.replace("~", fake_home) if path.startswith("~") else path
+        )
+
+        try:
+            tilde_dir = "~/test_tilde_dir"
+            func(tilde_dir)
+            expected_path = os.path.join(fake_home, "test_tilde_dir")
+            assert os.path.exists(
+                expected_path
+            ), "Tilde-expanded directory should be created"
+            print(f"✓ {name} Test 4 passed: Tilde expansion")
+        finally:
+            # Restore original expanduser
+            os.path.expanduser = original_expanduser
+
+
+if __name__ == "__main__":
+    _test_directory_creation_standalone()
+    _test_get_system_memory_standalone()
+    _test_load_class_standalone()
```

Comment:
1. please move this file to the `_common/tests/` directory with the other tests (somehow I didn't see this when Abrar added the file 🤦)
2. use `pytest` to run the tests, see other files for an example

Reply:
Ok, i raised a separate PR for refactoring existing tests - https://github.com/ray-project/ray/pull/53543
Once that lands, I can move the tests in this PR to `test_utils.py`.

Reply:
other PR is landed now 👍